build_tools/cmake/build_android.sh

```diff
@@ -29,22 +29,146 @@ else
   mkdir "${BUILD_ANDROID_DIR}"
 fi
 
-declare -a args=(
+declare -a common_args=(
   -G Ninja
   -B "${BUILD_ANDROID_DIR}"
   -DCMAKE_TOOLCHAIN_FILE="${ANDROID_NDK}/build/cmake/android.toolchain.cmake"
   -DANDROID_ABI="${ANDROID_ABI}"
   -DANDROID_PLATFORM=android-29
   -DIREE_HOST_BINARY_ROOT="${IREE_HOST_BINARY_ROOT}"
-  -DIREE_ENABLE_ASSERTIONS=ON
   -DIREE_BUILD_COMPILER=OFF
-  -DIREE_BUILD_TESTS=ON
   -DIREE_BUILD_SAMPLES=OFF
 )
 
-# Configure towards 64-bit Android 10, then build.
-"${CMAKE_BIN}" "${args[@]}"
+# Settings for the minimal build used for code size / c++ symbols checks.
+# See "### C++ symbols checks" section below.
+declare -a minimal_build_args=(
+  -DIREE_BUILD_TESTS=OFF
+  # A bunch of instrumentation is tied to IREE_ENABLE_ASSERTIONS - this isn't
+  # just about eliding assertions.
+  -DIREE_ENABLE_ASSERTIONS=OFF
+  # The Vulkan driver is C++.
+  -DIREE_HAL_DRIVER_VULKAN=OFF
+)
```

Comment:
There are other flags to set for minimal size builds. See https://github.com/iree-org/iree/blob/main/build_tools/cmake/build_runtime_small.sh and `IREE_SIZE_OPTIMIZED`, for example.

Reply:
I looked at these size-minimizing options.

There is a difference, even a tension, between minimizing code size by disabling parts of the code, and monitoring code size which requires enabling that code. Disabling code means adding blind spots in this C++ check.

The options that I retained here mirror what one would do at the moment in an Android build that doesn't necessarily need Vulkan and that is interested in avoiding the ~120 kB size increase from touching C++ on Android, but that would still like to remain casual, not super aggressive, in its size minimization.

For instance, `IREE_SIZE_OPTIMIZED` disables all the status message reporting code. That strikes me as aggressive, something that only the most size-conscious users would do. Definitely useful for embedded targets, but unlikely on Android.